src/components/SearchBox/onHighlightFactory.ts

```diff
@@ -31,7 +32,9 @@ export const getSkeleton = ({ geocoder }: GeocoderOption): Feature => {
 };
 
 export const onHighlightFactory =
-  (setPreview: (feature: unknown) => void) => (_: never, option: Option) => {
+  (setPreview: (feature: unknown) => void) =>
+  (_: never, option: Option, reason: AutocompleteHighlightChangeReason) => {
+    if (reason === 'touch') return;
```

Comment:
Can you please explain what was the issue here? I know it sometimes needed a double click, though I don't understand why this should help. (Though it helps! 🎉 😃 )

Reply:
The first click on the option runs the highlight callback. The second clicks then ran the actual click listener. 
Actually I also don't fully understand why instantly returning fixes this issue 😅 but it does and now we just return when the click was from a touch device.

Reply:
I see.. on click is like a hover on mobile. That makes sense! THX

> instantly returning fixes the issue

That almost looks like some mobile magic.. 😅 aka "if callback takes shorter than something, lets skip it .. apple is known for that 🙃